source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -670,25 +615,16 @@ bool CIrrDeviceSDL::run()
 				}
 #endif
 				irrevent.EventType = irr::EET_KEY_INPUT_EVENT;
-
-				if (isNoUnicodeKey(key))
-					irrevent.KeyInput.Char = 0;
-				else
-					irrevent.KeyInput.Char = SDL_event.key.keysym.sym;
-
 				irrevent.KeyInput.Key = key;
 				irrevent.KeyInput.PressedDown = (SDL_event.type == SDL_KEYDOWN);
 				irrevent.KeyInput.Shift = (SDL_event.key.keysym.mod & KMOD_SHIFT) != 0;
 				irrevent.KeyInput.Control = (SDL_event.key.keysym.mod & KMOD_CTRL ) != 0;
-
-				// SDL2 no longer provides unicode character in the key event so we ensure the character is in the correct case
-				// TODO: Unicode input doesn't work like this, should probably use SDL_TextInputEvent
-				bool convertCharToUppercase =
-					irrevent.KeyInput.Shift != !!(SDL_event.key.keysym.mod & KMOD_CAPS);
-
-				if (convertCharToUppercase)
-					irrevent.KeyInput.Char = irr::core::locale_upper(irrevent.KeyInput.Char);
-
+				// These keys are handled differently in CGUIEditBox.cpp (may become out of date!)
+				// Control key is used in special character combinations, so keep that too
+				// Pass through the keysym only then so no extra text gets input
+				irrevent.KeyInput.Char = 0;
+				if (mp.SDLKey == KEY_DELETE || mp.SDLKey == KEY_RETURN || mp.SDLKey == KEY_BACK || irrevent.KeyInput.Control)
+					irrevent.KeyInput.Char = mp.SDLKey;

```

Comment:
do sdl keys roughly correspond to chars or why does this work?

Reply:
It does not work. (I thought it worked because I was testing the wrong menu. Now it works.)